### ISSUE TITLE
Rounding error for large x values

### ISSUE THREAD

**Issue by ValentinKaisermayer** [MakieOrg/Makie.jl#1373]
```julia
y = cumsum(randn(1000))
x = 1:1000

f = Figure()
ax1 = Axis(f[1,1]) 
lines!(x, y)
ax1 = Axis(f[2,1]) 
lines!(x .+ Int(1e8), y)
f
```
![bug](https://user-images.githubusercontent.com/50108075/136530358-8779f341-28db-4a74-8693-040f9aed6015.png)

Makie v0.15.2
CairoMakie v0.6.5

**Comment by ffreyer** [comment]
See also #1196, #760, #931

**Comment by clamydo** [comment]
> We could also normalize the input to something like (-1, 1) before converting to Float32. Since all these issues are for 2D data it's probably fine to only do this when plotting to `Axis`.

Sounds reasonable. Scaling my time series this way before passing it to Makie and unscaling it for display via `axis.xtickformat` at least works. Perhaps this should be done internally before rendering?

Took me at least an hour pondering on my data before realising its an issue when plotting. In my humble and respectful opinion, this should have a higher priority to get fixed. It's damning for a plotting library, if it cannot represent data faithfully.

**Comment by asinghvi17** [comment]
Looks fixed with v0.21
![iTerm2 H8RgB5](https://github.com/MakieOrg/Makie.jl/assets/32143268/43384095-ba95-462e-876d-78af407de363)
